node/http.go

```diff
@@ -112,6 +113,73 @@ func makeRPCOverWebsocket(host string, port uint16, servers map[string]*jsonrpc.
 	return makeHTTPService(host, port, cors.Default().Handler(mux))
 }
 
+type ipcService struct {
+	handlers []*jsonrpc.Ipc
+}
+
+var _ service.Service = (*ipcService)(nil)
+
+// Run is a method of the ipcService struct that starts the execution of IPC handlers in parallel.
+// It takes context as a parameter to control the execution and returns an error if any handler fails.
+func (ipc *ipcService) Run(ctx context.Context) error {
+	errCh := make(chan error, len(ipc.handlers))
+	defer close(errCh)
+
+	var wg conc.WaitGroup
+	defer wg.Wait()
+
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
```

Comment:
If we use `context.WithCancelCause(ctx)` here, I think we can get rid of the error channel.

Instead of `if err := handler.Run(ctx); ... { errCh <- err } ` we use `cancel(handler.Run(ctx))`.

Instead of the `select` at the end of the function we can do something like `return context.Cause(ctx)`.

Reply:
Thanks for the tip. Wll check that out, never actually used `WithCancelCause` :laughing: